### PR TITLE
Managed kafka code samples

### PR DESCRIPTION
**Pull request by luongadam**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [n/a] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [n/a] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [x] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by m-strzelczyk — CHANGES_REQUESTED** [review]
I left some comments, only one is a potentially real issue with the code. Rest is more stylistic/organizational.

There are 2 more things to address:
1. You didn't provide `requirements.txt` in this PR in the `managedkafka/snippets` folder, so Koko doesn't run tests for this folder. Please provide `requirements.txt` to have Koko run the tests.
2. I personally like when my list and get methods return the values they're getting/listing. Making it so, provides users with clear indication with what kind of results they will deal with.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -92,6 +92,7 @@
 /composer/**/*                         @GoogleCloudPlatform/cloud-dpes-composer @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
 /pubsub/**/*                           @GoogleCloudPlatform/api-pubsub-and-pubsublite @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
 /pubsublite/**/*                       @GoogleCloudPlatform/api-pubsub-and-pubsublite @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
+/managedkafka/**/*                     @GoogleCloudPlatform/api-pubsub-and-pubsublite @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
```
[the remark this answers is not in the document]
Comment (luongadam) [reply .github/CODEOWNERS:95]: Yup, this is correct. All three products fall under the same team.

File: managedkafka/snippets/consumergroups/update_consumer_group.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        project_id, region, cluster_id, consumer_group_id
+    )
+
+    topic_metadata = managedkafka_v1.ConsumerTopicMetadata()
+    for partition, offset in partition_offsets.items():
+        partition_metadata = managedkafka_v1.ConsumerPartitionMetadata(offset=offset)
+        topic_metadata.partitions = {partition: partition_metadata}
+    consumer_group.topics = {
+        topic_path: topic_metadata,
+    }
```
[the remark this answers is not in the document]
Comment (luongadam) [reply managedkafka/snippets/consumergroups/update_consumer_group.py:57]: Yes this was intentional. Each partition index can only be mapped to a single offset value. With the loop, we can follow this logic for multiple partition indexes.

I updated the docstring for `partition_offsets`. Hopefully this clear things up.

Resolving this comment. Feel free to reopen!

File: managedkafka/snippets/topics/create_topic.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+
+def create_topic(
+    project_id: str,
+    region: str,
+    cluster_id: str,
+    topic_id: str,
+    partition_count: int,
+    replication_factor: int,
+    configs: dict[str, str],
```
[the remark this answers is not in the document]
Comment (luongadam) [reply managedkafka/snippets/topics/create_topic.py:27]: They can be found here https://kafka.apache.org/documentation/#topicconfigs. Added that to the docstring for configs.

File: managedkafka/snippets/consumergroups/update_consumer_group.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply managedkafka/snippets/consumergroups/update_consumer_group.py:57]: The similar part in Go is:

```go
partitionMetadata := make(map[int32]*managedkafkapb.ConsumerPartitionMetadata)
        for partition, offset := range partitionOffsets {
                partitionMetadata[partition] = &managedkafkapb.ConsumerPartitionMetadata{
                        Offset: offset,
                }
        }
```

In Java it's:

```java
Map<Integer, ConsumerPartitionMetadata> partitions =
    new HashMap<Integer, ConsumerPartitionMetadata>() {
      {
        for (Entry<Integer, Integer> partitionOffset : partitionOffsets.entrySet()) {
          ConsumerPartitionMetadata partitionMetadata =
              ConsumerPartitionMetadata.newBuilder()
                  .setOffset(partitionOffset.getValue())
                  .build();
          put(partitionOffset.getKey(), partitionMetadata);
        }
      }
    };
```


I think in Python, it should be:

```python
    topic_metadata = managedkafka_v1.ConsumerTopicMetadata()
    for partition, offset in partition_offsets.items():
        partition_metadata = managedkafka_v1.ConsumerPartitionMetadata(offset=offset)
        topic_metadata.partitions[partition] = partition_metadata
    consumer_group.topics = {
        topic_path: topic_metadata,
    }
```

[the remark this answers is not in the document]
Comment (luongadam) [reply managedkafka/snippets/consumergroups/update_consumer_group.py]: Ah good catch! Done.

**Review by m-strzelczyk — APPROVED** [review]

**Pull request merged**